sources/web/datalab/polymer/components/datalab-files/datalab-files.ts

```diff
@@ -93,10 +93,41 @@ class FilesElement extends Polymer.Element {
    * to initialize element state.
    */
   ready() {
+    // Must set this to true before calling super.ready(), because the latter will cause
+    // property updates that will cause _fetchFileList to be called first, we don't want
+    // that. We want ready() to be the entry point so it gets the user's last saved path.
+    this._fetching = true;
+
     super.ready();
 
-    // TODO: [yebrahim] The current path should be fetched from the server
-    // on initialization, in order to get the last saved user path
+    // Get the last startup path.
+    ApiManager.getUserSettings()
+      .then((settings: UserSettings) => {
+        if (settings.startuppath) {
+          let path = settings.startuppath;
+          if (path.startsWith(this.basePath)) {
+            path = path.substr(this.basePath.length);
+          }
+          // For backward compatibility with the current path format.
+          if (path.startsWith('/tree/')) {
+            path = path.substr('/tree/'.length);
+          }
+          this.currentPath = path;
+        }
+      })
+      .catch(() => {
+        this._fetching = false;
```

Comment:
You don't need to set _fetching to false here, because you are doing that in the following then block, which will always be executed after this catch block. If your intent is not to execute the following then block, you should either throw from this catch block, or move this catch block to the end of the chain.